### PR TITLE
fuzz: introduce initial fuzz targets

### PR DESCRIPTION
**Pull request by howardjohn**
These take raw bytes, and convert to Proto types then internal types.
For https://github.com/istio/ztunnel/issues/183

### CODE REVIEW

File: fuzz/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+target
```
Comment (hzxuzhonghu) [inline fuzz/.gitignore:1]: shouldn't this go to the root gitignore

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM except the gitignore

File: fuzz/.gitignore (hunk 1 above)
Comment (howardjohn) [reply fuzz/.gitignore:1]: it's for fuzz/target. This file is generated by `cargo fuzz`

Comment (hzxuzhonghu) [reply fuzz/.gitignore:1]: good to know

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**